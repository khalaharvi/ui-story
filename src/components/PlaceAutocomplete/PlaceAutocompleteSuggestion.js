import React from 'react';
import styled from 'react-emotion';
import PropTypes from 'prop-types';

const PlaceName = styled('span')({
  overflow: 'hidden',
  display: 'block',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
});

const PlaceAutocompleteSuggestion = ({ place }) => (
  <div><PlaceName>{place.name}</PlaceName></div>
);

PlaceAutocompleteSuggestion.propTypes = {
  place: PropTypes.shape({
    name: PropTypes.string.isRequired
  })
};

export default PlaceAutocompleteSuggestion;
