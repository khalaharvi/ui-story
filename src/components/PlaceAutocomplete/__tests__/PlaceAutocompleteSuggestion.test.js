import React from 'react';
import { shallow } from 'enzyme';

import PlaceAutocompleteSuggestion from '../PlaceAutocompleteSuggestion';

describe('<PlaceAutocompleteSuggestion />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <PlaceAutocompleteSuggestion
        place={{
          name: 'New York, NY'
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
