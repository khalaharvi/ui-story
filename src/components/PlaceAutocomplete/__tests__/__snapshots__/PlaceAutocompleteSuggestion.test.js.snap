// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaceAutocompleteSuggestion /> renders correctly 1`] = `
<div>
  <Styled(span)>
    New York, NY
  </Styled(span)>
</div>
`;
